File: Find the Prefix Common Array of Two Arrays.py
# https://leetcode.com/problems/find-the-prefix-common-array-of-two-arrays/?envType=daily-question&envId=2025-01-14
# Find the Prefix Common Array of Two Arrays

class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        
        n = len(A)
        seen_in_A = set()
        seen_in_B = set()
        prefix_common = []

        for i in range(n):
            seen_in_A.add(A[i])
            seen_in_B.add(B[i])
            common_count = len(seen_in_A.intersection(seen_in_B))
            prefix_common.append(common_count)

        return prefix_common

File: Find the Prefix Common Array of Two Arrays.js
// https://leetcode.com/problems/find-the-prefix-common-array-of-two-arrays/?envType=daily-question&envId=2026-05-20
// Find the Prefix Common Array of Two Arrays

/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number[]}
 */
var findThePrefixCommonArray = function(A, B) {

    const n = A.length;

    const seenInA = new Set();
    const seenInB = new Set();

    const prefixCommon = [];

    for (let i = 0; i < n; i++) {
        seenInA.add(A[i]);
        seenInB.add(B[i]);

        let commonCount = 0;

        for (const num of seenInA) {
            if (seenInB.has(num)) {
                commonCount++;
            }
        }

        prefixCommon.push(commonCount);
    }

    return prefixCommon;
};